Neos.ContentRepository/Classes/Domain/Context/Dimension/ContentDimension.php

```diff
@@ -0,0 +1,203 @@
+<?php
+namespace Neos\ContentRepository\Domain\Context\Dimension;
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+use Neos\ContentRepository\Domain\Context\Dimension\Exception\MissingContentDimensionValuesException;
+use Neos\Utility\Arrays;
+
+/**
+ * The content dimension domain model
+ */
+final class ContentDimension
+{
+    /**
+     * @var ContentDimensionIdentifier
+     */
+    protected $identifier;
+
+    /**
+     * An array of dimension values, indexed by their identifiers
+     *
+     * @var array|ContentDimensionValue[]
+     */
+    protected $values = [];
+
+    /**
+     * @var array|ContentDimensionValue[]
```

Comment:
indexed by "specialization" -> so you can answer questions like "what's the next-most generic value for the given value"